packages/devtools_app/lib/src/device_dialog.dart

```diff
@@ -31,6 +31,8 @@ class DeviceDialog extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    if (!serviceManager.connectedState.value.connected) return const SizedBox();
```

Comment:
nit: are we really showing the device dialog when not connected?
Consider instead just checking whether serviceManager.vm is null rather than the connected state so it is clear you are just avoiding showing content when that content isn't available.

Reply:
strangely enough, yes the dialog build method gets hit one last time when we disconnect the app. I spend a while trying to debug, but ended up just adding this check to be safe.